packages/mobx-state-tree/src/types/complex-types/model.ts

```diff
@@ -117,65 +119,78 @@ export type ModelCreationType<T extends ModelProperties> = {
 } &
     { [K in keyof OptionalProps<T>]?: ExtractC<T[K]> }
 
+export type ModelCreationType2<T extends ModelProperties, FixedC> = FixedOrOther<
+    FixedC,
+    ModelCreationType<T>
+>
+
 export type ModelSnapshotType<T extends ModelProperties> = { [K in keyof T]: ExtractS<T[K]> }
 
-export type ModelInstanceType<T extends ModelProperties, O, C, S> = {
+export type ModelSnapshotType2<T extends ModelProperties, FixedS> = FixedOrOther<
+    FixedS,
+    ModelSnapshotType<T>
+>
+
+export type ModelInstanceType<T extends ModelProperties, O, FixedC, FixedS> = {
     [K in keyof T]: ExtractIStateTreeNode<ExtractC<T[K]>, ExtractS<T[K]>, ExtractT<T[K]>>
 } &
     O &
-    IStateTreeNode<C, S>
+    IStateTreeNode<ModelCreationType2<T, FixedC>, ModelSnapshotType2<T, FixedS>>
 
 export interface ModelActions {
     [key: string]: Function
 }
 
-export interface IModelType<
-    PROPS extends ModelProperties,
-    OTHERS,
-    C = ModelCreationType<PROPS>,
-    S = ModelSnapshotType<PROPS>,
-    T = ModelInstanceType<PROPS, OTHERS, C, S>
-> extends IComplexType<C, S, T> {
+export interface IModelType<PROPS extends ModelProperties, OTHERS, FixedC, FixedS>
```

Comment:
FixedC/S are the overrides applied when using pre/postprocesssnapshot, they are false by default (rather than the whole blown type) to indicate that they are not overriden
I tried 'never' instead of false, but that wouldn't work well for some reason (maybe never never extends never? whatever)